### DISCUSSION TITLE
Support urql ssr prefetching without suspense

### DISCUSSION THREAD

**Discussion by danielr18** [pmndrs/jotai#1425] [Ideas]
In order to use async atoms without `Suspense`, we can wrap them with the `loadable` API.

The issue with the urql integration using the `loadable` API is that the resultAtom value is always a promise, so the loadable atom value will always be `{ state: 'loading' }` during SSR even if the data is prefetched.

If it's possible to get a sync snapshot during SSR (perhaps by passing a suspense: false arg), then resultAtom could be the prefetched result instead of a promise. I imagine something similar to react urql `useQuery` hook
https://github.com/FormidableLabs/urql/blob/bea39c2eb4a93669f5d1d4ee775dfc45cf56a39d/packages/react-urql/src/hooks/useQuery.ts#L99

FYI: SSR prefetching works with jotai query and `loadable` API.

**Comment by dai-shi** [comment]
Hi, thanks for opening this up.
This is slightly related with #1068 and I'm now thinking of redesigning query&urql integrations for various reasons:
- suspense oriented API should just return `Data`, but that doesn't cover all use cases.
- there's some inconsistency between jotai/query and jotai/urql API design, due to it.
- implementation is very hard each and not very maintainable. it should be a thin wrapper.
- the current versioning can confuse people. https://github.com/pmndrs/jotai/discussions/1312#discussioncomment-3329805

This will be a big change, and we should give migration path. It's on my action items. I'll ask for help when something is ready to play with.
